Competition2017/src/competition/subsystems/feeder/SideFeederSubsystem.java

```diff
@@ -0,0 +1,170 @@
+
+        atSpeedProp = propManager.createEphemeralProperty("Is" + side + "Feeder at speed?", false);
+        feederTalonError = propManager.createEphemeralProperty(side + "Feeder speed error", 0);
+        enableFeederLogging = propManager.createEphemeralProperty("Is " + side + "Feeder logging info?", false);
+
+        this.masterMotor = factory.getCANTalonSpeedController(motor);
+
+        setUpMotorTeam(masterMotor);
+        masterMotor.createTelemetryProperties(side + "Feeder master", propManager);
+    }
+
+    private void setUpMotorTeam(XCANTalon master) {
+
+        log.info("Configuring " + side + "motor");
+
+        master.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        master.reverseSensor(true);
+        master.setInverted(true);
+
+        log.info("Configuring voltages");
+
+        master.configNominalOutputVoltage(0,  -0);
+        master.configPeakOutputVoltage(12, -6);
+        master.setProfile(0);
+        master.setF(.099);
+
+        updateMotorPidValues(master);
+
+        master.setControlMode(TalonControlMode.Speed);
+        master.set(0);
+    }
+
+    private void updateMotorPidValues(XCANTalon motor) {
+        
+        motor.setP(p.get());
+        motor.setI(i.get());
+        motor.setD(d.get());
+        motor.setF(f.get());
+    }
+
+    public void setFeederPower(double power) {
+
+        masterMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
+        masterMotor.set(power);
+    }
+
+    public double getFeederPower() {
+
+        masterMotor.getOutputVoltage();
+        double inversionFactor = masterMotor.getInverted() ? -1 : 1;
+
+        return masterMotor.get() * inversionFactor;
+    }
+
+    public void setFeederTargetSpeed(double speed) {
+
+        masterMotor.ensureTalonControlMode(TalonControlMode.Speed);
+        feederTargetSpeed.set(speed);
+
+        updateMotorPidValues(masterMotor);
+        masterMotor.set(speed);
+    }
+
+    public double getSpeed() {
+
+        return masterMotor.getSpeed();
+    }
+
+    public boolean isAtSpeed() {
+
+        return Math.abs(getSpeed() - feederTargetSpeed.get()) <= feederSpeedThresh.get();
+    }
+
+    public double getFeederTargetSpeed() {
+
+        return feederTargetSpeed.get();
+    }
+
+    public void updateTelemetry() {
+
+        masterMotor.updateTelemetryProperties();        
+
+        atSpeedProp.set(isAtSpeed());
+
+        feederCurrentSpeed.set(getSpeed());
+
+        feederOutputPower.set(masterMotor.getOutputVoltage() / masterMotor.getBusVoltage());
+        feederTalonError.set(masterMotor.getClosedLoopError());
+
+        if(enableFeederLogging.get()){
+
+            double currentTime = Timer.getFPGATimestamp();
+
+            log.info(currentTime + "," 
+                 + feederOutputPower.get() + "," 
+                 + feederTalonError.get() + ","
+                 + feederCurrentSpeed.get());
+        }
+    }
+
+    public double getPowerStep(){
```

Comment:
@JohnGilb not sure we'll want/need the same power step here? I'm guessing this is the kind of thing we'd just tune in the dashboard?